File: Python/165. Compare Version Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        arr1 = version1.split(".")
        arr2 = version2.split(".")
        i = 0
        while i < min(len(arr1), len(arr2)):
            if int(arr1[i]) > int(arr2[i]):
                return 1
            elif int(arr1[i]) < int(arr2[i]):
                return -1
            i += 1
        if len(arr1) == len(arr2):
            return 0
        elif i == len(arr1):
            while i < len(arr2):
                if int(arr2[i]) > 0:
                    return -1
                i += 1
        elif i == len(arr2):
            while i < len(arr1):
                if int(arr1[i]) > 0:
                    return 1
                i += 1
        return 0

File: Java/165. Compare Version Numbers.java
package Java;

class Solution {
    public int compareVersion(String version1, String version2) {
        String[] arr1 = version1.split("\\.");
        String[] arr2 = version2.split("\\.");
        int i = 0;
        for (; i < Math.min(arr1.length, arr2.length); i++) {
            if (Integer.parseInt(arr1[i]) > Integer.parseInt(arr2[i])) {
                return 1;
            } else if (Integer.parseInt(arr1[i]) < Integer.parseInt(arr2[i])) {
                return -1;
            }
        }
        if (arr1.length == arr2.length) return 0;
        else if (i == arr1.length) {
            for (; i < arr2.length; i++) {
                if (Integer.parseInt(arr2[i]) > 0) return -1;
            }
        } else {
            for (; i < arr1.length; i++) {
                if (Integer.parseInt(arr1[i]) > 0) return 1;
            }
        }
        return 0;
    }
}